src/test/java/com/github/commonsrdf/dummyimpl/GraphImpl.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.commonsrdf.dummyimpl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.github.commonsrdf.api.BlankNodeOrIRI;
+import com.github.commonsrdf.api.Graph;
+import com.github.commonsrdf.api.IRI;
+import com.github.commonsrdf.api.RDFTerm;
+import com.github.commonsrdf.api.Triple;
+
+public class GraphImpl implements Graph {
+
+	private static final int TO_STRING_MAX = 10;
+	protected List<Triple> triples = new ArrayList<Triple>();
```

Comment:
Yes - it was - but for this to make sense then `Triple` and `IRI` and so on should implement `hashCode()` and `equals()` - that's a separate issue.   

As the javadoc for `Triple` does not require that, I do not implement it here - similarly `Graph.add()` is not required to do anything special on duplicates (and I test covering for both cases, e.g. `afterCount > beforeCount` after multiple add of same triple.